python/tests/test_tree_stats.py

```diff
@@ -4252,13 +4470,29 @@ def verify_sample_sets(self, ts, sample_sets, windows):
                 print("ts    ", sfs3)
             self.assertArrayAlmostEqual(sfs1, sfs2)
             self.assertArrayAlmostEqual(sfs1, sfs3)
+            assert np.all(sfs3 >= 0)
 
 
 class TestBranchAlleleFrequencySpectrum(
     TestAlleleFrequencySpectrum, TopologyExamplesMixin
 ):
     mode = "branch"
 
+    def generate_params(self, ts):
```

Comment:
Here is the main code testing the new time windows; this gets called on all the example tree sequences.

Reply:
(Note that it also determines which combinations of windows and sample sets get tested.)